modules/imap/hm-ews.php

```diff
@@ -924,9 +957,25 @@ protected function extract_mailbox($data) {
             }
             return $result;
         } elseif (is_object($data) && $data->Mailbox) {
-            return $data->Mailbox->getName() . ' <' . $data->Mailbox->getEmailAddress() . '>';
-        } elseif (is_object($data) && $data->getMailbox()) {
-            return $data->getMailbox()->getName() . ' <' . $data->getMailbox()->getEmailAddress() . '>';
+            if(is_array($data->Mailbox)) {
```

Comment:
I'm pretty sure the other open EWS PR has these updates and the ones below. Please reuse them by pushing your changes to that PR directly or wait for it to be merged and then update this one.